pootle/apps/pootle_translationproject/views.py

```diff
@@ -50,6 +50,51 @@ def admin_permissions(request, translation_project):
                        'translation_projects/admin/permissions.html', ctx)
 
 
+def get_sidebar_announcements_context(request, project, language):
+    announcements = []
+    new_cookie_data = {}
+    cookie_data = {}
+
+    if SIDEBAR_COOKIE_NAME in request.COOKIES:
+        json_str = unquote(request.COOKIES[SIDEBAR_COOKIE_NAME])
+        cookie_data = json.loads(json_str)
+
+    is_sidebar_open = cookie_data.get('isOpen', True)
+
+    def get_announcement(virtual_path):
+        try:
+            announcement = StaticPage.objects.live(request.user).get(
+                virtual_path=virtual_path,
+            )
+        except StaticPage.DoesNotExist:
+            pass
+        else:
+            announcements.append(announcement)
+
+            ann_mtime = dateformat.format(announcement.modified_on, 'U')
+            stored_mtime = cookie_data.get(virtual_path, None)
+
+            if ann_mtime != stored_mtime:
+                is_sidebar_open = True
+                new_cookie_data[virtual_path] = ann_mtime
+
+    get_announcement('announcements/projects/' + project.code)
```

Comment:
This is very unintuitive, with a function named `get_X()` programmers expect to have some value in return, but the function mutates an object which lives out of the function and returns nothing. Even more, also alters `is_sidebar_open` and `new_cookie_data`. I'd suggest to `get_announcement()` to do just what its name says, no more.

Also rather than building and passing a string, I'd rather pass `project=<project_code>, language=<language_code>` and build the string internally (use interpolation, no need for concatenation), that way the callers don't have to know or deal with the internal details of announcements.